nixos/modules/services/hardware/throttled.nix

```diff
@@ -4,10 +4,21 @@ with lib;
 
 let
   cfg = config.services.throttled;
+
+  customConfigFile =
+    if cfg.customConfig != null
+    then pkgs.writeText "lenovo_fix.conf" cfg.customConfig
+    else null;
 in {
   options = {
     services.throttled = {
       enable = mkEnableOption "fix for Intel CPU throttling";
```

Comment:
customConfig is usually called extraConfig across nixpkgs. Also if you set a default of "", you can get rid of all your "null" checks. Ifyou want to only optionally write the file, just check it the string is not null

Reply:
Good point! I've simplified the code!